src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -187,16 +196,19 @@ def wrap(*args, **kwargs):
                             df_ast_id = arg.df_ast_id
                             stmt_cache = arg.session._ast_batch._bind_stmt_cache
                             break
-                    df_transform_debug_trace = None
+                    df_transform_debug_trace = ""
                     try:
                         if (
                             _enable_dataframe_trace_on_error
                             and df_ast_id is not None
                             and stmt_cache is not None
                         ):
-                            df_transform_debug_trace = get_df_transform_trace_message(
+                            trace_result = get_df_transform_trace_message(
```

Comment:
I think it is easier to return empty string here https://github.com/snowflakedb/snowpark-python/blob/b75739c6b999b1b689d0652b7928d64c08583498/src/snowflake/snowpark/_internal/debug_utils.py#L181 to minimize the changes